mitmproxy/addons/xss.py

```diff
@@ -0,0 +1,391 @@
+from mitmproxy import ctx  # Used for logging information to mitmproxy
+from socket import gaierror, gethostbyname  # Used to check whether a domain name resolves
+from urllib.parse import urlparse  # Used to modify paths and queries for URLs
+import requests  # Used to send additional requests when looking for XSSs
+import re  # Used for pulling out the payload
+from html.parser import HTMLParser  # used for parsing HTML
+
+# The actual payload is put between a frontWall and a backWall to make it easy
+# to locate the payload with regular expressions
+frontWall = b"1029zxc"
+backWall = b"3847asd"
+payload = b"""s'd"ao<ac>so[sb]po(pc)se;sl/bsl\\"""
+fullPayload = frontWall + payload + backWall
+
+# A URL is a string starting with http:// or https:// that points to a website
+# A XSSDict is a dictionary with the following keys value pairs:
+#   - 'URL' -> URL
+#   - 'Injection Point' -> String
+#   - 'Exploit' -> String
+#   - 'Line' -> String
+# A SQLiDict is a dictionary with the following keys value pairs:
+#   - 'URL' -> URL
+#   - 'Injection Point' -> String
+#   - 'Regex' -> String
+#   - 'DBMS' -> String
+
+
+def getCookies(flow):
+    """ Return a dict going from cookie names to cookie values
+          - Note that it includes both the cookies sent in the original request and
+            the cookies sent by the server
+        Flow -> Dict """
+    # responseCookies is a list of tuples going from the cookie name to a SetCookie object
+    responseCookies = flow.response.cookies.fields
+    # requestCookies is a list of tuples going from the cookie name to the value
+    requestCookies = flow.request.cookies.fields
+    cookieDict = {}
+    for name, SC in responseCookies:
+        cookieDict[name] = SC.value
+    for name, value in requestCookies:
+        cookieDict[name] = value
```

Comment:
If the response sets a new value for cookie `"foo"`, you'll here overwrite it with the previous value from the request.
I'm not sure if you need the response cookies (the initial request worked without them as well), but you can probably simplify this to
 
```python
# untested
cookies = flow.request.cookies.copy()
cookies.update(flow.response.cookies)
return dict(cookies)
```